tools/bazar/fields/BazarField.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+namespace YesWiki\Bazar\Field;
+
+use Psr\Container\ContainerInterface;
+use YesWiki\Core\Service\TemplateEngine;
+
+abstract class BazarField
+{
+    protected $services;
+
+    protected $type;         // 0
+    protected $name;         // 1
+    protected $label;        // 2
+    protected $size;         // 3
+    protected $maxChars;     // 4
+    protected $default;      // 5
+    protected $required;     // 8
+    protected $searchable;   // 9
+    protected $hint;         // 10
+    protected $readAccess;   // 11
+    protected $writeAccess;  // 12
+    protected $semanticPredicate; // 14
+
+    // How the field is identified in the Bazar entry
+    protected $propertyName;
+
+    // Default values
+    protected const FIELD_TYPE = 0;
+    protected const FIELD_NAME = 1;
+    protected const FIELD_LABEL = 2;
+    protected const FIELD_SIZE = 3;
+    protected const FIELD_MAX_CHARS = 4;
+    protected const FIELD_DEFAULT = 5;
+    protected const FIELD_REQUIRED = 8;
+    protected const FIELD_SEARCHABLE = 9;
+    protected const FIELD_HINT = 10;
+    protected const FIELD_READ_ACCESS = 11;
+    protected const FIELD_WRITE_ACCESS = 12;
+    protected const FIELD_SEMANTIC_PREDICATE = 14;
+
+    public function __construct(array $values, ContainerInterface $services)
+    {
+        $this->services = $services;
+
+        $this->type = $values[self::FIELD_TYPE];
+        $this->name = $values[self::FIELD_NAME];
+        $this->label = $values[self::FIELD_LABEL];
+        $this->size = $values[self::FIELD_SIZE];
+        $this->maxChars = $values[self::FIELD_MAX_CHARS];
+        $this->default = $values[self::FIELD_DEFAULT];
+        $this->required = $values[self::FIELD_REQUIRED] == 1;
+        $this->searchable = $values[self::FIELD_SEARCHABLE];
+        $this->hint = $values[self::FIELD_HINT];
+        $this->readAccess = $values[self::FIELD_READ_ACCESS];
+        $this->writeAccess = $values[self::FIELD_WRITE_ACCESS];
+        $this->semanticPredicate = $values[self::FIELD_SEMANTIC_PREDICATE];
+
+        // By default, the entry ID is the field name
+        $this->propertyName = $values[self::FIELD_NAME];
+    }
+
+    // Render the edit view of the field. Check ACLS first
+    public function renderInputIfPermitted($entry)
+    {
+        // Safety checks, must be run before every renderInput
+        if( !$this->canEdit($entry) ) return '';
+
+        return $this->renderInput($entry);        
+    }
+
+    // Format input values before save
+    public function formatValuesBeforeSave($entry)
+    {
+        return [$this->propertyName => $this->getValue($entry)];
+    }
+
+    // Render the show view of the field
+    public function renderStatic($entry)
+    {
+        return $this->render("@bazar/fields/{$this->type}.twig", [
```

Comment:
```suggestion
         return empty($this->getValue($entry)) ? '' : $this->render("@bazar/fields/{$this->type}.twig", [
```

Pour garder le même comportement qu'avant, je propose de ne pas faire le rendu pour les valeurs vides.